client/src/components/overview/QuantitySelector.jsx

```diff
@@ -1,37 +1,63 @@
 import React from 'react';
 
-const QuantitySelector = (props) => {
-  console.log('Rendering Quantity Selector');
-  // console.log('props.maxQuantity', props.maxQuantity);
-  const maxQuantityCapped = Math.min(props.maxQuantity, 15);
-  const quantitiesAvailable = props.maxQuantity && maxQuantityCapped > 0;
+class QuantitySelector extends React.Component {
+  constructor(props) {
+    super(props);
 
-  const resetToDefaultSelected = () => {
+    this.onChange = this.onChange.bind(this);
+    this.getAttributes = this.getAttributes.bind(this);
+
+    this.state = {
+      currentQuantity: 0
+    }
+  }
+
+  static defaultProps = {
+    maxQuantityLimit: 15,
+  }
+
+  componentDidUpdate() {
+    const quantitiesAreAvailable = this.quantitiesAreAvailable(this.props.maxQuantity, this.props.maxQuantityLimit);
+    console.log('componentDidUpdate - quantitiesAreAvailable:', quantitiesAreAvailable);
+    if (this.state.currentQuantity === 0 && quantitiesAreAvailable) {
+      console.log('Resetting current quantity to: ', 1);
+      this.setState({ currentQuantity: 1 });
+    } else if (this.state.currentQuantity > 0 && !quantitiesAreAvailable) {
+      console.log('Resetting current quantity to: ', 0);
+      this.setState({ currentQuantity: 0 });
+    }
+  }
+
+  getMaxQuantityCapped(maxQuantity, maxQuantityLimit) {
+    return Math.min(maxQuantity, maxQuantityLimit);
+  }
+
+  quantitiesAreAvailable(maxQuantity, maxQuantityLimit) {
+    return !!(maxQuantity && (this.getMaxQuantityCapped(maxQuantity, maxQuantityLimit) > 0));
+  }
+
+  resetToDefaultSelected() {
+    console.log('Resetting quantity selected')
     const options = document.querySelectorAll('#po-quantity option');
```

Comment:
Oh boy ... is this the weirdness everybody was talking about with this select element?
I'm sure you know this is anti-pattern in terms of React, but I certainly don't know a "React way" to do this.

Reply:
Yep. I didn't like it, but for the sake of time & lack of prior understanding, it had to be done :-o